src/Generator/Program.fs

```diff
@@ -0,0 +1,189 @@
+    
+    let extensions = Parsing.getExtensions test
+    printfn "Extension count: %d" extensions.Length
+
+    let functionToExtensionMapper =
+        extensions
+        |> Array.Parallel.collect(fun extension ->
+            extension.functions |> Array.Parallel.map(fun func -> extension.name, func))
+        |> Array.groupBy snd
+        |> Array.Parallel.map(fun (func, extensions) -> func, extensions |> Array.Parallel.map fst)
+        |> Map.ofArray
+
+    let enumCaseToExtensionMapper =
+        extensions
+        |> Array.Parallel.collect(fun extension ->
+            extension.enumCases |> Array.Parallel.map(fun enumCase -> extension.name, enumCase))
+        |> Array.groupBy snd
+        |> Array.Parallel.map(fun (enumCase, extensions) -> enumCase, extensions |> Array.Parallel.map fst)
+        |> Map.ofArray
+            
+
+    let typecheckedFunctions =
+        TypeMapping.looslyTypedFunctionsToTypedFunctions enumMap functions
+        |> Array.Parallel.map Formatting.PrintReady.formatTypedFunctionDeclaration
+        |> Array.Parallel.collect(fun func ->
+            match functionOverloads |> Map.tryFind func.actualName with
+            | Some overload ->
+                overload.overloads
+                |> Array.Parallel.map(fun currOverload ->
+                    { func with
+                        prettyName = overload.alternativeName |> Option.defaultValue func.prettyName
+                        parameters = currOverload.parameters |> Array.map Formatting.PrintReady.formatTypeTypeParameterInfo
+                        retType = currOverload.retType |> Formatting.PrintReady.formatTypeInfo }
+                )
+            | None -> func |> Array.singleton
+        )
+
+    let prettyEnumGroups =
+        enums
+        |> Array.Parallel.map Formatting.PrintReady.formatEnumGroup
+
+    let prettyEnumGroupMap =
+        prettyEnumGroups
+        |> Array.Parallel.map(fun group ->
+            group.groupName, group
+        )
+        |> Map.ofArray
+
+    printfn "overall correct function specifications: %d" typecheckedFunctions.Length
+    let openGlSpecVersions = Aggregator.getEnumCasesAndCommandsPerVersion test
+    let basePath = options.pathToOutputDirectory
+
+    let getShortTagForOpenGlVersion (openGl: RawOpenGLSpecificationDetails) =
+        let str = sprintf "%M" openGl.versionNumber
+        if openGl.version.Contains "ES" then str + "_ES"
+        else str
+
+    let getPathForOpenGlVersion (openGl: RawOpenGLSpecificationDetails) =
+        basePath </> (getShortTagForOpenGlVersion openGl)
+
+    basePath
+    |> Directory.CreateDirectory
+    |> ignore
+
+    Formatting.generateDummyTypes
+    |> fun content -> 
+        File.WriteAllText(basePath </> dummyTypesFileName + ".cs", content)
+
+    let inline writeToFile (openGl: RawOpenGLSpecificationDetails) topic content =
+        let pathToFile = getPathForOpenGlVersion openGl
+        pathToFile
+        |> Directory.CreateDirectory
```

Comment:
I feel the amount of resemblance between the pipeline of functionToExtensionManager & enumCaseToExtensionMapper is quite high. Let's refactor this :)

Reply:
s you're right